src/Nri/Ui/Carousel/V2.elm

```diff
@@ -0,0 +1,206 @@
+module Nri.Ui.Carousel.V2 exposing
+    ( viewWithCombinedControls, viewWithPreviousAndNextControls
+    , viewWithTabControls, LabelledBy(..), Role(..)
+    )
+
+{-| Patch changes:
+
+  - added new carousel APIs (with tabbed controls/previous and next controls, and combined controls) using [W3C guidelines](https://www.w3.org/WAI/ARIA/apg/patterns/carousel/)
+
+@docs viewWithCombinedControls, viewWithPreviousAndNextControls
+@docs viewWithTabControls, LabelledBy, Role
+
+-}
+
+import Accessibility.Styled.Aria as Aria
+import Css exposing (..)
+import Html.Styled as Html exposing (..)
+import Html.Styled.Attributes as Attrs exposing (css)
+import TabsInternal.V2 as TabsInternal
+
+
+{-| Type which represents the type of aria label which will be used
+`LabelledByIdOfVisibleLabel` will point to an existing element id on the DOM
+`LabelledByAccessibleLabelOnly` will be a label of the element
+-}
+type LabelledBy
+    = LabelledByIdOfVisibleLabel String
+    | LabelledByAccessibleLabelOnly String
+
+
+{-| `Role`, which can be either [Group](https://w3c.github.io/aria/#group) or [Region](https://w3c.github.io/aria/#region)
+-}
+type Role
+    = Group
+    | Region
+
+
+{-| Builds a carousel with previous and next controls
+Returns:
+`controls`: previous and next buttons element
+`slides` the container with the carousel contents
+`containerAttributes` attributes that should be used on the parent div of both the button and slides elements
+-}
+viewWithPreviousAndNextControls :
+    { selected : id
+    , panels :
+        List
+            { id : id
+            , slideHtml : Html msg
+            , labelledBy : LabelledBy
+            }
+    , viewPreviousButton : Html msg
+    , viewNextButton : Html msg
+    , labelledBy : LabelledBy
+    , role : Role
+    }
+    ->
+        { viewPreviousButton : Html msg
+        , viewNextButton : Html msg
+        , slides : Html msg
+        , containerAttributes : List (Attribute msg)
+        }
+viewWithPreviousAndNextControls config =
+    { viewPreviousButton = config.viewPreviousButton
+    , viewNextButton = config.viewNextButton
+    , slides =
+        List.map
+            (\panel ->
+                Html.div
+                    [ Attrs.attribute "role" "group"
```

Comment:
Not a big deal, so feel free to ignore, but you should be able to use `Role.group`